job/dependency_resolver.go

```diff
@@ -27,29 +26,27 @@ var (
 const InterJobDependencyNameSections = 2
 
 type dependencyResolver struct {
-	jobSpecRepo   store.JobSpecRepository
-	jobSourceRepo store.JobSourceRepository
-	pluginService service.PluginService
-
-	// TODO: will be deprecated along with Resolve method deprecation
-	projectJobSpecRepoFactory ProjectJobSpecRepoFactory
+	projectSpecRepo store.ProjectRepository
+	jobSpecRepo     store.JobSpecRepository
+	jobSourceRepo   store.JobSourceRepository
+	pluginService   service.PluginService
 
 	externalDependencyResolver ExternalDependencyResolver
 }
 
 // NewDependencyResolver creates a new instance of Resolver
 func NewDependencyResolver(
+	projectSpecRepo store.ProjectRepository,
 	jobSpecRepo store.JobSpecRepository,
 	jobSourceRepo store.JobSourceRepository,
 	pluginService service.PluginService,
-	projectJobSpecRepoFactory ProjectJobSpecRepoFactory,
 	externalDependencyResolver ExternalDependencyResolver,
 ) DependencyResolver {
 	return &dependencyResolver{
+		projectSpecRepo:            projectSpecRepo,
```

Comment:
yes, we can name it that way. the reason I chose `projectSpecRepo` is so that it is similar to `jobSpecRepo` (avoiding confusion between `models.JobSpec` and `models.Job`)

Reply:
`projectSpecRepo` is removed after the newer update, as it is not required anymore